account_move_group_restriction/tests/test_account_move_group_restriction.py

```diff
@@ -0,0 +1,173 @@
+                "account_type": "asset_current",
+                "company_id": cls.company.id,
+            }
+        )
+        cls.account_restricted = cls.env["account.account"].create(
+            {
+                "name": "Restricted Account",
+                "code": "RES001",
+                "account_type": "asset_current",
+                "company_id": cls.company.id,
+                "security_group_ids": [
+                    Command.set([cls.group_restricted_account_management.id])
+                ],
+            }
+        )
+        cls.journal = cls.env["account.journal"].create(
+            {
+                "name": "Miscellaneous",
+                "code": "TEST MISC",
+                "type": "general",
+                "company_id": cls.company.id,
+            }
+        )
+        cls.move_public = cls._create_move(cls.account_public, cls.account_public)
+        cls.move_restricted = cls._create_move(
+            cls.account_restricted, cls.account_restricted
+        )
+        cls.move_mixed = cls._create_move(cls.account_restricted, cls.account_public)
+
+    @classmethod
+    def _create_move(cls, debit_account, credit_account):
+        Move = cls.env["account.move"]
+        move = Move.create(
+            {
+                "move_type": "entry",
+                "journal_id": cls.journal.id,
+                "company_id": cls.company.id,
+                "line_ids": [
+                    Command.create(
+                        {
+                            "name": "Debit line",
+                            "account_id": debit_account.id,
+                            "debit": 100,
+                            "credit": 0.0,
+                        }
+                    ),
+                    Command.create(
+                        {
+                            "name": "Credit line",
+                            "account_id": credit_account.id,
+                            "debit": 0.0,
+                            "credit": 100,
+                        }
+                    ),
+                ],
+            }
+        )
+        move.action_post()
+        return move
+
+    def test_account_security_group_ids_computed_from_accounts(self):
+        self.assertFalse(self.move_public.account_security_group_ids)
+        self.assertEqual(
+            self.move_restricted.account_security_group_ids,
+            self.group_restricted_account_management,
+        )
+        self.assertEqual(
+            self.move_mixed.account_security_group_ids,
+            self.group_restricted_account_management,
+        )
+
+    def test_basic_user_cannot_see_restricted_moves(self):
+        Move_basic = self.env["account.move"].with_user(self.user_basic)
+        visible_public = Move_basic.search([("id", "=", self.move_public.id)])
+        visible_restricted = Move_basic.search([("id", "=", self.move_restricted.id)])
+        visible_mixed = Move_basic.search([("id", "=", self.move_mixed.id)])
```

Comment:
A minor point. To be consistent with variables in `test_basic_user_cannot_see_restricted_move_lines`.
```suggestion
        public_move = Move_basic.search([("id", "=", self.move_public.id)])
        restricted_move = Move_basic.search([("id", "=", self.move_restricted.id)])
        mixed_move = Move_basic.search([("id", "=", self.move_mixed.id)])
```